plaidml/bridge/openvino/ops/prior_box.cpp

```diff
@@ -0,0 +1,447 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset4.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;  // NOLINT[build/namespaces]
+using ngraph::opset4::PriorBox;
+
+namespace PlaidMLPlugin {
+
+struct LocalConfigs {
+  int H;
+  int W;
+  int IH;
+  int IW;
+  int num_of_prior;
+  int min_element_size;
+  float step;
+  float step_x;
+  float step_y;
+  float offset;
+  std::vector<float> aspect_ratios;
+  std::vector<float> aspect_ratios_scale_size;
+  std::vector<float> variance;
+  std::vector<float> min_size;
+};
+
+// Preprocess data with some attributes
+void prepareConfig(LocalConfigs& conf, const ngraph::op::PriorBoxAttrs& attrs, const Context& ctx) {
+  int& H = conf.H;
+  int& W = conf.W;
+  int& IH = conf.IH;
+  int& IW = conf.IW;
+  auto* input_shape_ngraph_op = ngraph::as_type<ngraph::op::Constant>(ctx.layer->get_input_node_ptr(0));
+  std::vector<int> input_shape = input_shape_ngraph_op->get_vector<int>();
+  H = input_shape[0];
+  W = input_shape[1];
+  auto* image_shape_ngraph_op = ngraph::as_type<ngraph::op::Constant>(ctx.layer->get_input_node_ptr(1));
+  std::vector<int> image_shape = image_shape_ngraph_op->get_vector<int>();
+  IH = image_shape[0];
+  IW = image_shape[1];
+
+  conf.num_of_prior = ngraph::op::PriorBox::number_of_priors(attrs);
+
+  std::vector<float>& aspect_ratios = conf.aspect_ratios;
+  aspect_ratios.push_back(1.0f);
+  for (const auto& aspect_ratio : attrs.aspect_ratio) {
+    bool exist = false;
+    for (const auto existed_value : aspect_ratios) exist |= std::fabs(aspect_ratio - existed_value) < 1e-6;
+
+    if (!exist) {
+      aspect_ratios.push_back(aspect_ratio);
+      if (attrs.flip) {
+        aspect_ratios.push_back(1.0f / aspect_ratio);
+      }
+    }
+  }
+  for (float ar : aspect_ratios) {
+    if (std::fabs(ar - 1.0f) < 1e-6) {
+      continue;
+    }
+    conf.aspect_ratios_scale_size.push_back(std::sqrt(ar));
+  }
```

Comment:
if (std::fabs(ar - 1.0f) >= 1e-6) {
      conf.aspect_ratios_scale_size.push_back(std::sqrt(ar));
    }
    Is this the same?

Reply:
Yes, but we use "< 1e-6" as  a common way to show float compare, may not change as it will broke the tradition

Reply:
I see. We can keep it as it is.